lighthouse-core/gather/gather-runner.js

```diff
@@ -216,6 +216,28 @@ class GatherRunner {
     return new LHError(LHError.errors.CHROME_INTERSTITIAL_ERROR);
   }
 
+  /**
+   * Returns an error if we try to load a non-HTML page.
+   * @param {LH.Artifacts.NetworkRequest|undefined} mainRecord
+   * @return {LH.LighthouseError|undefined}
+   */
+  static getNonHtmlError(mainRecord) {
+    // MIME types are case-insenstive
+    const HTML_MIME_REGEX = /^text\/html$/i;
+
+    // If we never requested a document, there's no doctype error, let other cases handle it.
+    if (!mainRecord) return undefined;
+
+    // mimeType is determined by the browser, we assume Chrome is determining mimeType correctly,
+    // independently of 'Content-Type' response headers, and always sending mimeType if well-formed.
+    if (mainRecord.mimeType || mainRecord.mimeType === '') {
```

Comment:
`mainRecord.mimeType` is initialized as an empty string and set from `response.mimeType` which apparently is always a string, never null or undefined. So I think this can be refactored to just:

```
if (!HTML_MIME_REGEX.test(mainRecord.mimeType)) {
  return new LHError(LHError.errors.NON_HTML, {mimeType: mainRecord.mimeType});
}
```

(removing the outer if statement)

Reply:
thanks for clarifying, I was unsure if the string from the response could be overwritten in the browser with a non-string / null / undefined, I'll make sure to fix this!